ngraph/core/src/op/max_pool.cpp

```diff
@@ -223,3 +313,54 @@ shared_ptr<Node> op::v8::MaxPool::clone_with_new_inputs(const OutputVector& new_
                                     m_index_element_type,
                                     m_axis);
 }
+
+bool op::v8::MaxPool::has_evaluate() const {
+    NGRAPH_OP_SCOPE(v8_MaxPool_has_evaluate);
+    switch (get_input_element_type(0)) {
+    case ngraph::element::i32:
+    case ngraph::element::i64:
+    case ngraph::element::u32:
+    case ngraph::element::u64:
+    case ngraph::element::f16:
+    case ngraph::element::f32:
+        return true;
+    default:
+        break;
+    }
+    return false;
+}
+
+bool op::v8::MaxPool::evaluate(const HostTensorVector& outputs, const HostTensorVector& inputs) const {
+    NGRAPH_OP_SCOPE(v8_MaxPool_evaluate);
+    return evaluate_maxpool(outputs, inputs);
+}
+
+bool op::v8::MaxPool::evaluate_maxpool(const HostTensorVector& outputs, const HostTensorVector& inputs) const {
+    NGRAPH_OP_SCOPE(v8_MaxPool_evaluate_maxpool);
```

Comment:
it can be a body of evaluate method without an introduction of additional private method.